pkg/operation/botanist/health_check.go

```diff
@@ -488,12 +488,19 @@ func (b *HealthChecker) CheckMonitoringControlPlane(
 		return exitCondition, nil
 	}
 
-	statefulSetList, err := statefulSetLister.StatefulSets(namespace).List(monitoringSelector)
+	requiredMonioringStatefulSets := sets.NewString(common.PrometheusStatefulSetName)
```

Comment:
How about defining a method called `RequiredMonitoringStatefulSets` (btw, typo in your variable name) that computes the required monitoring stateful sets based on a given `gardenv1beta1.Shoot`?